fenix/app/src/main/java/org/mozilla/fenix/compose/Menu.kt

```diff
@@ -62,14 +68,51 @@ private fun Menu(
                 .background(color = FirefoxTheme.colors.layer2)
                 .then(modifier),
         ) {
+            val hasCheckedItems = menuItems.any { it.isChecked }
+
             for (item in menuItems) {
+                val checkmarkModifier = if (hasCheckedItems) {
+                    Modifier.selectable(
+                        selected = item.isChecked,
+                        role = Role.Button,
+                        onClick = {
+                            onDismissRequest()
+                            item.onClick()
+                        },
+                    )
+                } else {
+                    Modifier
+                }
+
                 DropdownMenuItem(
-                    modifier = Modifier.testTag(item.testTag),
+                    modifier = Modifier
+                        .testTag(item.testTag)
+                        .align(alignment = Alignment.CenterHorizontally)
+                        .then(checkmarkModifier),
                     onClick = {
                         onDismissRequest()
                         item.onClick()
                     },
                 ) {
+                    if (hasCheckedItems) {
+                        if (item.isChecked) {
+                            Icon(
+                                painter = painterResource(id = R.drawable.mozac_ic_checkmark_24),
+                                modifier = Modifier
+                                    .size(24.dp),
+                                contentDescription = null,
+                                tint = FirefoxTheme.colors.iconPrimary,
+                            )
+
+                            Spacer(modifier = Modifier.size(width = 12.dp, height = 0.dp))
+                        } else {
+                            Spacer(
+                                modifier = Modifier
+                                    .size(width = 36.dp, height = 24.dp),
+                            )
+                        }
```

Comment:
```suggestion
                        } else {
                            Spacer(
                                modifier = Modifier
                                    .size(24.dp),
                            )
                        }
                        
                        Spacer(modifier = Modifier.width(12.dp))
```

Let's do something like this for the similar reason in anticipation of the grid system.

Reply:
done